core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/MemberAccessorFactory.java

```diff
@@ -6,38 +6,59 @@
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 import org.optaplanner.core.api.domain.common.DomainAccessType;
 import org.optaplanner.core.api.solver.SolverFactory;
 import org.optaplanner.core.impl.domain.common.ReflectionHelper;
 import org.optaplanner.core.impl.domain.common.accessor.gizmo.GizmoMemberAccessorFactory;
 
-public class MemberAccessorFactory {
+public final class MemberAccessorFactory {
+
+    private static final MemberAccessorFactory INSTANCE = new MemberAccessorFactory();
```

Comment:
Knowing Geoffrey, he would not like this. This creates a leak. The factory needs to be limited to something that gets garbage collected, such as solver config or something like that.

The current impl keeps the factory forever, regardless of whether the solver is long gone from memory or not.

Reply:
Fair point; see the alternative.